storage/statedb/database.go

```diff
@@ -431,13 +442,15 @@ func (db *Database) node(hash common.Hash) node {
 			logger.Error("node from cached trie node fails to be decoded!", "err", err)
 		}
 	}
+
+	// Retrieve the node from the state cache if available
 	db.lock.RLock()
 	node := db.nodes[hash]
 	db.lock.RUnlock()
-
 	if node != nil {
 		return node.obj(hash)
 	}
+
```

Comment:
Changes in L#437 and L#453 seems unnecessary.

Reply:
I added lines for readability.